### PR TITLE
add supported provider type

### PR DESCRIPTION
**Pull request by weibaohui**
add supported provider type

### CODE REVIEW

File: docs/en/developers/references/traits/route.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ Name | Type | Description | Notes
 **Domain** | **string** | specify your host url for this app | [ default to (empty) ]
 **Issuer** | **string** | specify your certificate issue  | [default to no tls]
 **Rules** | [**[]RouteRules**](#routerules) |  | [optional] 
+**Provider** | **string** | ingress controller name ,we support **nginx**,**[contour](#https://github.com/projectcontour/contour)**  | [empty means using nginx-ingress]
```
Comment (zzxwill) [inline docs/en/developers/references/traits/route.md:32]:
```suggestion
**Provider** | **string** | ingress controller name ,we support **nginx**,**[contour](#https://github.com/projectcontour/contour)**  | [default to nginx, which means using nginx-ingress]
```

Let's specify the default value first:)

**Comment by resouer** [comment]
Note that the user of vela doesn't know what is contour or nginx ingress. So there won't be a 'type' field in trait spec.

Instead, the provider of ingress is a platform admin's concern that should be decided by:

1. trait definition, and be configurable by vela install config
2. or, auto discover by whether contour or nginx ingress is available (default provider should be defined in this case)

1. is simplest and 2. is favorable, leave to @wonderflow for final decision

**Comment by wonderflow** [comment]
@resouer yes, I think we could adopt the first solution as a quick fix, that make the provider field configured in traitdefinition template.

**Comment by wonderflow** [comment]
closing as we have automatically generate this doc

**Pull request closed without merging**